lib/haml/util.rb

```diff
@@ -207,13 +207,13 @@ def unescape_interpolation(str, escape_html = nil)
         if escapes % 2 == 1
           res << "\##{char}"
         else
-          interpolated = if char == '{'
+          interpolated = if char == '{'.freeze
             balance(scan, ?{, ?}, 1)[0][0...-1]
           else
             scan.scan(/\w+/)
           end
-          content = eval('"' + interpolated + '"')
-          content.prepend(char) if char == '@' || char == '$'
+          content = eval("\"#{interpolated}\"")
```

Comment:
See https://github.com/haml/haml/pull/961/commits/42f4eaa4cb8339bbefeecfa415169531a6d2c0b3 for justification, summary is that it saves a string allocation each time this code is called for roughly the same performance.